### PR TITLE
Add mongodb adapter to mongo build

### PR DESCRIPTION
**Pull request by enleur**
Use mongodb adapter in related builds

Related to https://github.com/sonata-project/SonataDoctrineMongoDBAdminBundle/pull/183
Fixes https://github.com/sonata-project/SonataDoctrineMongoDBAdminBundle/pull/186

PS I didn't check that it is working properly. Could you?

### CODE REVIEW

File: project/.travis/before_install_test.sh.twig [hunk 1]
```diff
@@ -14,6 +14,10 @@ fi
 composer self-update --stable
 sed --in-place "s/\"dev-master\":/\"dev-${TRAVIS_COMMIT}\":/" composer.json
 
+{% if 'mongodb' in services %}
+if [ "$MONGODB_VERSION" != "" ]; then pecl install -f mongodb-$MONGODB_VERSION; fi
```
Comment (greg0ire) [inline project/.travis/before_install_test.sh.twig]: Please use `--force` instead of `-f`, it's more explicit. Also, why do you need this option?

Comment (Enleur) [inline project/.travis/before_install_test.sh.twig]: In case when the adapter was installed with an outdated version

**Comment by Enleur** [comment]
@Soullivaneuh ping

**Comment by core23** [comment]
We need this for the exporter bundle too https://github.com/sonata-project/exporter/issues/113

File: project/.travis.yml.twig [hunk 2]
```diff
@@ -64,6 +64,10 @@ matrix:
 {% endfor %}
     - php: '{{ php|last }}'
       env: SYMFONY_DEPRECATIONS_HELPER=0
+{% if 'mongodb' in services %}
```
Comment (Soullivaneuh) [inline project/.travis.yml.twig]: This is a too specific modification. Please  not this is a template for a all the projects.

[This part](https://github.com/sonata-project/dev-kit/blob/8e1af1e5d91ed9f3b8fa7ca0eec26417655ba12b/config/dev-kit.yml#L45) should be modified instead.

Comment (Enleur) [inline project/.travis.yml.twig]: I don't get how [this](https://github.com/sonata-project/dev-kit/blob/8e1af1e5d91ed9f3b8fa7ca0eec26417655ba12b/config/dev-kit.yml#L45) should be used

File: project/.travis/before_install_test.sh.twig [hunk 3]
```diff
@@ -14,6 +14,10 @@ fi
 composer self-update --stable
 sed --in-place "s/\"dev-master\":/\"dev-${TRAVIS_COMMIT}\":/" composer.json
 
+{% if 'mongodb' in services %}
+pecl install --force mongodb-stable
+composer require "alcaeus/mongo-php-adapter=^1.0.0" --ignore-platform-reqs
```
Comment (Soullivaneuh) [inline project/.travis/before_install_test.sh.twig]: * This should be installed only on PHP7, should not be?
* You should add the `--no-update` option.

Comment (Soullivaneuh) [inline project/.travis/before_install_test.sh.twig]: BTW, is the `^1.0.0` constraint really necessary?

Comment (Enleur) [inline project/.travis/before_install_test.sh.twig]: > BTW, is the ^1.0.0 constraint really necessary?

not really

Comment (Enleur) [inline project/.travis/before_install_test.sh.twig]: Done

**Comment by core23** [comment]
Please rebase @Enleur

**Comment by Enleur** [comment]
@core23 done

File: config/projects.yml [hunk 4]
```diff
@@ -138,13 +138,13 @@ doctrine-extensions:
 doctrine-mongodb-admin-bundle:
   branches:
     master:
-      php: ['5.6']
+      php: ['5.3', '5.4', '5.5', '5.6', '7.0']
       services: [mongodb]
```
Comment (core23) [inline config/projects.yml]: We don't support old php versions in the latest master

File: config/projects.yml [hunk 5]
```diff
@@ -138,13 +138,13 @@ doctrine-extensions:
...
     master:
-      php: ['5.6']
+      php: ['5.6', '7.0']
       services: [mongodb]
       versions:
         symfony: ['2.8', '3.0', '3.1']
         sonata_admin: ['3']
     3.x:
-      php: ['5.3', '5.4', '5.5', '5.6']
+      php: ['5.3', '5.4', '5.5', '5.6', '7.0']
```
Comment (core23) [inline config/projects.yml]: Same here

File: project/.travis/before_install_test.sh.twig [hunk 6]
```diff
@@ -14,6 +14,13 @@ fi
 composer self-update --stable
 sed --in-place "s/\"dev-master\":/\"dev-${TRAVIS_COMMIT}\":/" composer.json
 
+
+if [ ${TRAVIS_PHP_VERSION} '>=' '7.0' ]; then
+    {% if 'mongodb' in services %}
+    pecl install --force mongodb-stable
+    composer require "alcaeus/mongo-php-adapter" --ignore-platform-reqs --no-update
```
Comment (Soullivaneuh) [inline project/.travis/before_install_test.sh.twig]: Instead of this, please play with the `packages` of `dev-kit.yml` file.

Comment (Enleur) [inline project/.travis/before_install_test.sh.twig]: not sure that i get what do you mean

Comment (Soullivaneuh) [inline project/.travis/before_install_test.sh.twig]: I mean [this](https://github.com/sonata-project/dev-kit/blob/c5b400abff42957e8252c4390fc8423887df65cc/config/dev-kit.yml#L45) and the usage [here](https://github.com/sonata-project/dev-kit/blob/c5b400abff42957e8252c4390fc8423887df65cc/config/projects.yml).

**Comment by core23** [comment]
Do we still need this or are there any update for travis @enleur ?

**Pull request closed without merging**